ironic_tempest_plugin/tests/scenario/ironic_standalone/test_basic_ops.py

```diff
@@ -436,3 +438,175 @@ class BaremetalRedfishIPxeWholediskHttpLink(
     @utils.services('network')
     def test_ip_access_to_server(self):
         self.boot_and_verify_node()
+
+
+class BaremetalIdracWSManDeployNode(bsm.BaremetalStandaloneScenarioTest):
+
+    mandatory_attr = ['driver']
+    api_microversion = '1.31'  # to set the deploy_interface
+    driver = 'idrac'
+    image_ref = CONF.baremetal.whole_disk_image_ref
+    boot_interface = 'ipxe'
+
+    @classmethod
+    def skip_checks(cls):
+        super(BaremetalIdracWSManDeployNode, cls).skip_checks()
+        if not CONF.baremetal_feature_enabled.ipxe_enabled:
+            skip_msg = ("Skipping this TC since ipxe is disabled")
+            raise cls.skipException(skip_msg)
+
+    @utils.services('image', 'network')
+    @decorators.idempotent_id('b467889b-aba4-4696-9e6f-4ef60a3f78a4')
+    def test_deploy_node(self):
+        self.boot_and_verify_node()
+
+
+class BaremetalIdracWSManDeployNodepxe(bsm.BaremetalStandaloneScenarioTest):
+
+    mandatory_attr = ['driver']
+    api_microversion = '1.31'  # to set the deploy_interface
+    driver = 'idrac'
+    image_ref = CONF.baremetal.whole_disk_image_ref
+    boot_interface = 'pxe'
+
+    @classmethod
+    def skip_checks(cls):
+        super(BaremetalIdracWSManDeployNodepxe, cls).skip_checks()
+        if CONF.baremetal_feature_enabled.ipxe_enabled:
+            skip_msg = ("Skipping the Test case since ipxe is enabled")
+            raise cls.skipException(skip_msg)
+
+    @utils.services('image', 'network')
+    @decorators.idempotent_id('20c0f2f4-723b-4d1e-a4f2-bea11ffb1513')
+    def test_deploy_node(self):
+        self.boot_and_verify_node()
+
+
+class BaremetalIdracRedfishDeployNode(bsm.BaremetalStandaloneScenarioTest):
```

Comment:
Why cant we inherit from "BaremetalIdracWSManDeployNode" instead of writing whole test case in a class.
Its just power and management interface which we are changing.

Reply:
Done